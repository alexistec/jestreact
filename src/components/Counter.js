import React,{useState} from 'react';


const Counter = () => {

    const [counter,setCounter] = useState(0);


    const increase = () => {
        setCounter(counter +1);
    }

    const decrease = () =>{
        setCounter(counter - 1);
    }

    return(
        <div>
            <h1>{counter}</h1>
            <button onClick={increase}>Increase</button>
            <button onClick={decrease}>Decrease</button>
        </div>
    )

}

export default Counter;