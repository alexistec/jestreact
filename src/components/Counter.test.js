import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import Counter from './Counter';

configure({
    adapter:new Adapter()
})


describe('Testing component <Counter/>',()=>{
    let wrapper;
    beforeEach(()=>{
        wrapper = shallow(<Counter/>)
    })

    test('Validate to match snapshot',()=>{
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    test('Validate botton operation',()=>{
        wrapper.find('button').first().simulate('click')
        expect(wrapper.find('h1').text()).toBe('0')
    })
})