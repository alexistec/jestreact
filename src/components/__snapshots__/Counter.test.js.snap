// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component <Counter/> Validate to match snapshot 1`] = `
<div>
  <h1>
    0
  </h1>
  <button
    onClick={[Function]}
  >
    Increase
  </button>
  <button
    onClick={[Function]}
  >
    Decrease
  </button>
</div>
`;
